app/src/org/commcare/android/models/NodeEntityFactory.java

```diff
@@ -34,6 +40,8 @@ public Detail getDetail() {
         getDetail().populateEvaluationContextVariables(nodeContext);
         //return new AsyncEntity<TreeReference>(detail.getFields(), nodeContext, data);
 
+        nodeContext.addFunctionHandler(hereFunctionHandler);

```

Comment:
I would consider writing: 
`nodeContext.addFunctionHandler(EntitySelectActivity.hereFunctionHandler);`
which would allow you to eliminate the `hereFunctionHandler` class field.